### PR TITLE
Add support for linear filtering of floating point textures.

### PR DESCRIPTION
**Pull request by bagnell**
Textures can now have any of the linear options for the minification and magnification filter when the extensions are available. Otherwise, the filtering defaults to nearest.

This was pulled from the hdr branch.

CHANGES.md was not updated. All of the modifications are at the Renderer level.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @bagnell!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/float-linear/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by bagnell** [comment]
Bump.

File: Specs/Renderer/CubeMapSpec.js [hunk 1]
```diff
@@ -396,6 +481,94 @@ defineSuite([
...
+            expect({
+                context : context,
+                fragmentShader : fs,
+                uniformMap : uniformMap,
+                epsilon : 1
+            }).contextToRender(color.toBytes());
+        }
+    });
+
+    it('creates a cube map with half floating-point textures and linear filtering', function() {
```
Comment (lilleyse) [inline Specs/Renderer/CubeMapSpec.js]: It looks like `creates a cube map with half floating-point textures` and `creates a cube map with half floating-point textures and linear filtering` are actually testing the opposite thing, should the titles be swapped?

File: Specs/Renderer/CubeMapSpec.js [hunk 2]
```diff
@@ -396,6 +481,94 @@ defineSuite([
...
+            'uniform samplerCube u_texture;' +
+            'void main() { gl_FragColor = textureCube(u_texture, normalize(vec3(1.0, 1.0, 0.0))); }';
+
+        var uniformMap = {
+            u_texture : function() {
+                return cubeMap;
+            }
+        };
+
+        if (!context.textureFloatLinear) {
```
Comment (lilleyse) [inline Specs/Renderer/CubeMapSpec.js]: Should this be `textureHalfFloatLinear`?

File: Source/Renderer/Context.js [hunk 3]
```diff
@@ -276,11 +276,15 @@ define([
...
         this._depthTexture = !!getExtension(gl, ['WEBGL_depth_texture', 'WEBKIT_WEBGL_depth_texture']);
-        this._textureFloat = !!getExtension(gl, ['OES_texture_float']);
-        this._textureHalfFloat = !!getExtension(gl, ['OES_texture_half_float']);
         this._fragDepth = !!getExtension(gl, ['EXT_frag_depth']);
         this._debugShaders = getExtension(gl, ['WEBGL_debug_shaders']);
 
+        this._textureFloat = !!getExtension(gl, ['OES_texture_float']);
+        this._textureHalfFloat = !!getExtension(gl, ['OES_texture_half_float']);
+
+        this._textureFloatLinear = !!getExtension(gl, ['OES_texture_float_linear']);
```
Comment (pjcozzi) [inline Source/Renderer/Context.js]: Does anything need to be taken into account for WebGL 1 vs. 2?

**Comment by pjcozzi** [comment]
Looks good at quick glance.

File: Source/Renderer/Context.js (hunk 3 above)
Comment (bagnell) [reply Source/Renderer/Context.js]: I'm assuming that the extension is still required for WebGL 2. I couldn't find anything in the spec that says it isn't allowed in WebGL 2; however, I do see the extension for `OES_texture_float_linear` under the WebGL 2 tab at webglreport.com. Also, I don't see the extension for `OES_texture_half_float_linear` in the WebGL 2 tab, but I do see it under the WebGL 1 tab.

Comment (lilleyse) [reply Source/Renderer/Context.js]: According to MDN it is both a WebGL 1 and 2 extension.
https://developer.mozilla.org/en-US/docs/Web/API/OES_texture_float_linear

Comment (bagnell) [reply Source/Renderer/Context.js]: And `OES_texture_half_float_linear` is gone under WebGL 2 and it's included with `OES_texture_float_linear`. (Similar to what they did for floating-point color buffer attachments)

**Comment by bagnell** [comment]
@lilleyse @pjcozzi Updated.

File: Source/Renderer/Context.js (hunk 3 above)
Comment (lilleyse) [reply Source/Renderer/Context.js]: Is https://developer.mozilla.org/en-US/docs/Web/API/OES_texture_half_float_linear inccorect?

**Comment by lilleyse** [comment]
Looks good, thanks @bagnell.

**Pull request merged**